mlflow/models/utils.py

```diff
@@ -97,18 +93,27 @@ def _is_scalar(x):
                 "(pandas.DataFrame, numpy.ndarray, dict, list), "
                 "got {}".format(type(input_example))
             )
+
         example_filename = "input_example.json"
-        self.data = input_example.to_dict(orient="split")
-        # Do not include row index
-        del self.data["index"]
-        if all(input_example.columns == range(len(input_example.columns))):
-            # No need to write default column index out
-            del self.data["columns"]
-        self.info = {
-            "artifact_path": example_filename,
-            "type": "dataframe",
-            "pandas_orient": "split",
-        }
+        if isinstance(input_example, pd.DataFrame):
```

Comment:
I think we should do it the other way - we should check if the input is tensors (i.e. a numpy array or a dict of numpy arrays) otherwise proceed to store dataframe. Note that we do support e.g. dict of scalars, or a list as valid examples. This is so that uses can easily handcraft examples.